example-hooks/copy-admin-config/post-cluster-init/copy-admin-config.yaml

```
@@ -0,0 +1,16 @@
+# Create .kube directory for vscode user
+- name: Create .kube directory for vscode user
+  delegate_to: localhost
+  run_once: true
+  ansible.builtin.file:
+    path: /home/vscode/.kube
+    state: directory
+    mode: '0700'
+
+- name: Copy admin kubeconfig to vscode user
+  delegate_to: "{{ first_kube_control_plane }}"
+  run_once: true
+  ansible.builtin.fetch:
+    src: /etc/kubernetes/admin.conf
+    flat: true
+    dest: /home/vscode/.kube/config
```

This will copy the kube config file to the user the playbook is running as, rather then a hard coded user.
```suggestion
- name: Copy file kubeconfig from first control plane node
  become: true
  delegate_to: "{{ first_kube_control_plane }}"
  register: kubernetes_admin_conf
  run_once: true
  ansible.builtin.slurp:
    src: /etc/kubernetes/admin.conf

- name: Create ~/.kube directory
  become: false
  delegate_to: localhost
  run_once: true
  ansible.builtin.file:
    dest: ~/.kube
    mode: "0700"
    state: directory

- name: Copy admin kubeconfig to ~/.kube directory
  become: false
  delegate_to: localhost
  run_once: true
  ansible.builtin.copy:
    content: "{{ kubernetes_admin_conf.content | b64decode }}"
    dest: ~/.kube/config
    mode: "0600"
```